src/js/libgeo/GeoOps.js

```diff
@@ -215,6 +234,25 @@ geoOps.HorizontalLine.updatePosition = function(el) {
     putStateComplexVector(param); // copy param
     el.homog = General.withUsage(param, "Line");
 };
+geoOps.HorizontalLine.getRandomMove = function(el) {
+    var rcomplex = function() {
+        var fact = 0.1;
+        return CSNumber.complex((Math.random() - 0.5) * fact, (Math.random() - 0.5) * fact);
+    };
+
+    var p = List.cross(el.homog, List.ex);
+    p = List.normalizeMax(p);
+
+    var rvec = List.turnIntoCSList([CSNumber.zero, rcomplex(), rcomplex()]);
+    rvec = List.scalmult(p.value[2], rvec);
+
+    var move = List.add(rvec, p);
+    var res = {
+        type: "mouse",
+        value: move
+    };
+    return res;
+};
```

Comment:
This should get the job done, but I can't help wondering whether there is a simpler approach using `"homog"` instead of `"mouse"` as the movement type. Can't we just add `[0, rcomplex(), rcomplex()]` to `el.homog`? If not, under what circumstances would this be problematic?

The code has trouble if the line is the line at infinity, e.g. after an explicit assignment to the `homog` field. Then `p.value[2]` is zero, so `rvec` is null. Furthermore `move` equals `List.ex` which makes the resulting line undefined.

Both comments above apply to `VerticalLine` as well.